json_yaml_files/prepare_files.py

```diff
@@ -0,0 +1,184 @@
+import re
+import json
+import sys
+import yaml
+import argparse
+from pathlib import Path
+
+
+class BIDStoNDAConfigGenerator:
+    def __init__(self):
+        # Set BIDS configuration
+        bids.config.set_option('extension_initial_dot', True)
+
+        # Parse command-line arguments
+        parser = argparse.ArgumentParser(description='Generate YAML and JSON files for a BIDS dataset.')
+        parser.add_argument('-s', '--source', type=str, required=True, help='Path to the BIDS dataset directory')
+        parser.add_argument('-d', '--destination', type=str, required=True, help='Path to save the generated YAML and JSON files')
+        args = parser.parse_args()
+        self.source_path = args.source
+        self.destination_path = Path(args.destination)
+
+        # Initialize BIDS layout
+        self.layout = BIDSLayout(self.source_path, derivatives=True)
+
+        # Define constants
+        self.A = 'image03'
+        self.pattern = r'^(?P<A>[^_]+)_(?P<X>[^.]+)\.(?P<Y>[^.]+)\.(?P<Z>[^.]+)\..+$'
+        self.scan_type = {
+            'anat': {
+                'mprage': 'MR structural (MPRAGE)',
+                't1w': 'MR structural (T1)',
+                'pd': 'MR structural (PD)',
+                'fspgr': 'MR structural (FSPGR)',
+                'fsip': 'MR structural (FISP)',
+                't2w': 'MR structural (T2)',
+                'pd_t2': 'MR structural (PD, T2)',
+                'b0_map': 'MR structural (B0 map)',
+                'b1_map': 'MR structural (B1 map)',
+                'flash': 'MR structural (FLASH)',
+                'mp2rage': 'MR structural (MP2RAGE)',
+                'tse': 'MR structural (TSE)',
+                't1w_t2w': 'MR structural (T1, T2)',
+                'mpnrage': 'MR structural (MPnRAGE)'
+            },
+            'pet': {
+                'pet': 'PET'
+            }
+        }
+        self.image_modality = {
+            'pet': 'PET',
+            'anat': 'MRI',
+            'func': 'MRI',
+            'dwi': 'MRI',
+            'fmap': 'MRI'
+        }
+        
+        # Set output directories
+        self.json_dir = self.destination_path / 'prepared_jsons'
+        self.yaml_dir = self.destination_path / 'prepared_yamls'
+        self.json_dir.mkdir(parents=True, exist_ok=True)
+        self.yaml_dir.mkdir(parents=True, exist_ok=True)
+
+    def extract_file_name_components(self, file_name):
+        match = re.match(self.pattern, file_name)
+        return match.groups() if match else (None, None, None, None)
+
+
+    def prepare_json_contents(self, file_name, files):
+        json_contents = {}
+        _, X, _, _ = self.extract_file_name_components(file_name)
+
+        sub_path_prefix = {
+            'derivatives': 'derivatives',
+            'inputs': 'sub-'
+        }
+        
+        for file_path in files:
+            sub_path = file_path[file_path.find(sub_path_prefix.get(X, '')):] if X in sub_path_prefix else file_path
```

Comment:
This line is hard to read. Please separate it out into an easier-to-understand multiple lines.